google/cloud/spanner_v1/transaction.py

```diff
@@ -100,7 +105,10 @@ def begin(self):
         database = self._session._database
         api = database.spanner_api
         metadata = _metadata_with_prefix(database.name)
-        txn_options = TransactionOptions(read_write=TransactionOptions.ReadWrite())
+        if read_only:
+            txn_options = TransactionOptions(read_only=TransactionOptions.ReadOnly())
```

Comment:
This will begin a read-only transaction on this connection. It is important to realize (and maybe also document) a couple of things when using a read-only transaction:
1. A read-only transaction will only read data at a specific point in time during the entire transaction. That means that when a user starts a read-only transaction, they will never see any new changes to the database until the read-only transaction has been closed and a new one begins. So if I start a new read-only transaction at time T1, and someone else executes and commits a read/write transaction at T2, the read-only transaction will still only see the data at T1 and not the data from T2. This can be surprising for many users, and it should be pointed out that most users who want to just read data should probably use autocommit mode, and not read-only transactional mode.
2. Read-only transactions cannot be committed or rolled back. You must therefore check the type of transaction that is in use when `commit` or `rollback` is called on the connection, and only do something with the transaction if it is a read/write transaction. (And of course return the session to the pool if it is a read-only transaction.)

Reply:
Okay, I've added changes for item 2 and a system test for item 1.
Looks like we need to find a proper place for mentioning all of it. I don't see any docs for connection API...

Reply:
Thanks, this looks great. I've added a couple of suggestion for documentation in the code.

Also, there is one more thing that we could simplify for read-only transactions: A read-only transaction will never be aborted by Cloud Spanner, so keeping track of a running checksum for the returned result sets is not necessary.  We could therefore skip that when we see that the transaction is in read-only mode.